### PR TITLE
Add Russian button translations

### PR DESCRIPTION
**Pull request by zenseii**
Usually we want more translated strings per translation PR, but since this feature of button translation is completely new we want to show it off as soon as possible.

Translations kindly provided by @Branikolog.

### CODE REVIEW

**Comment by ihhub** [comment]
@zenseii , thank you so much for this valuable improvement!

File: files/lang/ru.po [hunk 1]
```diff
@@ -25,16 +25,16 @@ msgstr ""
 msgid ""
 "BATTLE\n"
 "ONLY"
-msgstr ""
+msgstr "БИТВА"
 
 msgid "GIFT"
-msgstr ""
+msgstr "ОТДАТЬ"
```
Comment (a1exsh) [inline files/lang/ru.po:31]: Может быть "ДАР"?

Comment (zenseii) [reply files/lang/ru.po:31]: Yes, @Branikolog gave me that as an option too. You can change it to this in your PR if you feel that is better.

Comment (Branikolog) [reply files/lang/ru.po:31]: Hi, @a1exsh .
I think "ДАР" is a bit not informative here. Looks like some sort of abbreviation. More clear would be "В ДАР", but one single word looks more pleasant in my personal point.

Comment (a1exsh) [reply files/lang/ru.po:31]: I mean...  the player can just click the button to see what it does — nothing bad can happen :)

Comment (Alucard648) [reply files/lang/ru.po:31]: "ДАРИТЬ"

Comment (Branikolog) [reply files/lang/ru.po:31]: The best text here could have been "ПОДАРИТЬ" or "ПЕРЕДАТЬ", but the button size doesn't allow such long words.
"ДАРИТЬ" also doesn't fit well, as the majority of game buttons contain words in perfect tense, not continuous: "ЗАКРЫТЬ", but not "ЗАКРЫВАТЬ"

Comment (a1exsh) [reply files/lang/ru.po:31]: Мое последнее предложение: `ДАНЬ` :)

Comment (Branikolog) [reply files/lang/ru.po:31]: > Мое последнее предложение: `ДАНЬ` :)

:D
Дань - это подразумевает принуждение и неотвратимость данного действия. А механика является добровольной.)
Я согласен, то "Отдать" тоже не самый идеальный вариант, но это, на мой взгляд, лучшее, что влазит в данную кнопку.

**Comment by a1exsh** [comment]
Doesn't seem to work for me, though:
image

Am I missing something?  Built from commit 27ec7781

**Comment by Branikolog** [comment]
Hi, @a1exsh 
I've just checked the latest commit ver., everything works fine.
Did you restarted the application, after you switched to Russian?

**Comment by zenseii** [comment]
> Doesn't seem to work for me, though:
> 
> Am I missing something?


Did you recompile the .mo file with the string translated?

**Comment by a1exsh** [comment]
> Did you restarted the application, after you switched to Russian?

Hm, not sure — will try again :)

> Did you recompile the .mo file with the string translated?

Will double-check, thanks for the suggestions!

**Comment by a1exsh** [comment]
@Branikolog @zenseii all is fine, it's just me who forgot the workflow after a long break.  Thanks!

**Comment by zenseii** [comment]
> @Branikolog @zenseii all is fine, it's just me who forgot the workflow after a long break. Thanks!

BTW @a1exsh just fyi we haven't added the whole Cyrillic alphabet yet so if you try to use any character that hasn't been added then the English translation will show up. You can see what letters have been added here:
https://github.com/ihhub/fheroes2/pull/5903

They're not perfect but considering I almost have no experience with that alphabet it could've been worse :P

**Comment by a1exsh** [comment]
@zenseii awesome!  Ultimately, we should probably render all buttons ourselves, if the selected translation is different from the game's resources.  Right now it's a mix of English/Russian or German/Russian for me, depending on the installation.

**Comment by zenseii** [comment]
> @zenseii awesome! Ultimately, we should probably render all buttons ourselves, if the selected translation is different from the game's resources. Right now it's a mix of English/Russian or German/Russian for me, depending on the installation.

Yep, that's the endgoal to have all buttons be replaceable. Inevitably more letters will need to be added to the button fonts but hopefully not too many.

**Pull request merged**